src/breedsDb.js

```diff
@@ -18,13 +18,63 @@ class BreedRepository {
         return this.breeds;
     }
 
+    set setBreeds(breedsCopy) {
+        this.breeds = breedsCopy;
+    }
+
+    findIndexById(id) {
+        const breedsCopy = this.getAllBreeds; 
+        const foundBreed = breedsCopy.findIndex((breed) => {
+            return breed.breedId === id;       
+        });
+        return foundBreed;  // foundBreed is either -1 or a valid index
+    }
+
     getBreedById (id) {
         const breedsCopy = this.getAllBreeds; 
-        const idMatchArray = breedsCopy.filter((breed) => { 
+        const idMatchArray = breedsCopy.filter((breed) => {  // could use findIndexById instead of array.filter
             return breed['breedId'] == id; 
         })
         return idMatchArray[0];
     }
+
+    updateBreedById (id, object) {
+        const breedsCopy = this.getAllBreeds; 
+        const index = this.findIndexById(id); // 
+        if (index === -1) {
+            console.log(`index not present: ${index}`);
+            return null;
+        } else {
+            for (let key in object) {
+                if (breedsCopy[index][key] !== object[key]) {
```

Comment:
style nit: consider creating local variables for `breedsCopy[index]` and `object[key]`, as this will reduce the mental overhead required to read the code.